src/code/audio/audio_load.c

```diff
@@ -1,161 +1,2160 @@
+    if (sample == NULL) {
+        return;
+    }
+
+    slowLoad->sample = *sample;
+    sample->sampleAddr = slowLoad->ramAddr;
+    sample->medium = MEDIUM_RAM;
+}
+
+void AudioLoad_ProcessSlowLoads(s32 resetStatus) {
+    AudioSlowLoad* slowLoad;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.slowLoads); i++) {
+        slowLoad = &gAudioContext.slowLoads[i];
+        switch (gAudioContext.slowLoads[i].status) {
+            case LOAD_STATUS_LOADING:
+                if (slowLoad->medium != MEDIUM_UNK) {
+                    osRecvMesg(&slowLoad->msgqueue, NULL, OS_MESG_BLOCK);
+                }
+
+                if (resetStatus != 0) {
+                    slowLoad->status = LOAD_STATUS_DONE;
+                    continue;
+                }
+            case LOAD_STATUS_START:
+                slowLoad->status = LOAD_STATUS_LOADING;
+                if (slowLoad->bytesRemaining == 0) {
+                    AudioLoad_FinishSlowLoad(slowLoad);
+                    slowLoad->status = LOAD_STATUS_DONE;
+                    *slowLoad->isDone = 1;
+                } else if ((s32)slowLoad->bytesRemaining < 0x400) {
+                    if (slowLoad->medium == MEDIUM_UNK) {
+                        size_t size = slowLoad->bytesRemaining;
+
+                        AudioLoad_DmaSlowCopyUnkMedium(slowLoad->curDevAddr, slowLoad->curRamAddr, size,
+                                                       slowLoad->unkMediumParam);
+                    } else {
+                        AudioLoad_DmaSlowCopy(slowLoad, slowLoad->bytesRemaining);
+                    }
+                    slowLoad->bytesRemaining = 0;
+                } else {
+                    if (slowLoad->medium == MEDIUM_UNK) {
+                        AudioLoad_DmaSlowCopyUnkMedium(slowLoad->curDevAddr, slowLoad->curRamAddr, 0x400,
+                                                       slowLoad->unkMediumParam);
+                    } else {
+                        AudioLoad_DmaSlowCopy(slowLoad, 0x400);
+                    }
+                    slowLoad->bytesRemaining -= 0x400;
+                    slowLoad->curRamAddr += 0x400;
+                    slowLoad->curDevAddr += 0x400;
+                }
+                break;
+        }
+    }
+}
+
+void AudioLoad_DmaSlowCopy(AudioSlowLoad* slowLoad, size_t size) {
+    Audio_InvalDCache(slowLoad->curRamAddr, size);
+    osCreateMesgQueue(&slowLoad->msgqueue, &slowLoad->msg, 1);
+    AudioLoad_Dma(&slowLoad->ioMesg, OS_MESG_PRI_NORMAL, OS_READ, slowLoad->curDevAddr, slowLoad->curRamAddr, size,
+                  &slowLoad->msgqueue, slowLoad->medium, "SLOWCOPY");
+}
+
+void AudioLoad_DmaSlowCopyUnkMedium(s32 devAddr, s32 ramAddr, size_t size, s32 arg3) {
+}
+
+s32 AudioLoad_SlowLoadSeq(s32 seqId, u8* ramAddr, s8* isDone) {
+    AudioSlowLoad* slowLoad;
+    AudioTable* seqTable;
+    size_t size;
+
+    if (seqId >= gAudioContext.numSequences) {
+        *isDone = 0;
```

Comment:
```suggestion
        *isDone = false;
```
boolean?

Reply:
Leaving this as is as this poorly-named var is more of a status instead of a bool